app/controllers/users.js

```diff
@@ -0,0 +1,69 @@
+'use strict';
+
+const bcrypt = require('bcrypt');
+const errors = require('../errors');
+const userService = require('../services/users');
+const sessionManager = require('./../services/sessionManager');
+const isAlphaNumeric = require('validate.io-alphanumeric');
+
+exports.create = (req, res, next) => {
+  const saltRounds = 10;
+
+  const user = req.body
+    ? {
+        firstname: req.body.firstname,
+        lastname: req.body.lastname,
+        password: req.body.password,
+        email: req.body.email
+      }
+    : {};
+
+  if (!isAlphaNumeric(user.password)) {
+    return next(errors.invalidPasswordFormat);
+  }
+
+  if (user.password.length < 8) {
```

Comment:
With a regular expression you can validate length and wether its alphanumeric with one if statement